dhall-docs/README.md

```diff
@@ -97,6 +60,377 @@ dhall-docs --input . --output-link ${OTHER_LINK}
+(i.e. `{-|`), that comment will be considered invalid: `dhall-docs` will not
+render it on the documentation _and_ a warning will be logged on the console.
+The following:
+
+```dhall
+  {-| foo
+      bar -}
+```
+
+although it's a valid Dhall block comment, it's invalid for `dhall-docs`.
+
+If you write a comment (either block or single-line comments) after a valid
+`dhall-docs` comment, it will be ignored. Wheter they were valid `dhall-docs`
+comments or not. On the two following examples:
+
+```dhall
+{-|
+  foo -}
+-- bar
+```
+
+```dhall
+{-|
+  foo
+-}
+
+{-|
+-}
+```
+
+the extracted text will only contain `foo`.
+
+To end, if the `|` character isn't on the opening brace, `dhall-docs` will
+ignore the comment without logging on the console. This let you separate your
+own internal block comments from the ones that you want `dhall-docs` to output
+in the final markup.
+
+
+This is an example of an actual file documented using block comments:
+
+```dhall
+{-|
+# My awesome package
+
+This is the header of my package
+
+* a list item in **bold**
+  - a sub item in _italic_
+  - a [link](https://example.com)
+    * a `code` quote
+
+
+A new paragaph
+
+    an indented code block
+-}
+let myAwesomeFunction {- this will be ignored on the documentation -}
+               = \(x : Natural) -> x + 1
+
+in
+{
+  {-|
+  foo
+      bar
+          baz
+  -}
+  myAwesomeFunction
+}
+```
+
+### Line comments
+
+Normal Dhall single-line comments starts with `--` and ends at a newline. For
+example:
+
+```dhall
+-- single line comment
+0
+```
+
+A `dhall-docs` valid single-line comment starts with `--|` and a single whitespace
+character. For example:
+
+```dhall
+--| foo
+```
+
+You can span your documentation in several single-line comments by writing `--` and two
+(2) whitespaces:
+
+```dhall
+--| foo
+--  bar
+--      baz
+```
+
+The `--` prefix defines the indentation for any line. That is, on the above
+example `dhall-docs` will extract the following contents:
+
+```
+foo
+bar
+    baz
+```
+
+If you would like to force an empty line on the documentation output, place an
+empty single line comment in that place. On this example:
+
+```dhall
+--| foo
+--
+--  bar
+```
+
+`dhall-docs` will extract:
+
+```dhall
+foo
+
+bar
+```
+
+The empty `--` line doesn't need the two (2) whitespaces.
+
+If between two single-line comments there is one (1) or more empty lines, the
+comments after those will be ignored. On this example:
+
+
+```dhall
+--| foo
+--  bar
+
+--  baz
+```
+
+will only capture this text:
+
+```
+foo
+bar
+```
+
+Also note that in a set of consecutive single line comments where the `|` is
+not on the first one, the single-line comments before that one will be
+ignored on the rendered documentation. On this example:
+
+```dhall
+--  foo
+--  bar
+--| baz
+--  qux
+```
+
+the following text will be extracted:
+
+```dhall
+baz
+qux
+```
+
+The set of `--` lines needs to be aligned, meaning that a set of lines like this:
```

Comment:
"aligned" is slightly vague. Is this aligned?

```
	--| foo
 -- bar
```

What does it mean when before the `--|` there are [fullwidth characters](https://en.wikipedia.org/wiki/Halfwidth_and_fullwidth_forms)?

For simplicity, I think we can demand that each `--` is preceded by the same number of characters (if that's well-defined?!). Or we could give up the alignment requirement again…

Reply:
I agree with that definition of alignment. I'd say that will be enough for almost all cases.

I don't have a problem with no enforcing the alignment, but following @Gabriel439 comment ([here](https://github.com/dhall-lang/dhall-haskell/pull/1929#discussion_r456917946)) we should start with something more strict and make it more flexible depending on user requests.